File: 프로그래머스/0/120876. 겹치는 선분의 길이/겹치는 선분의 길이.java
class Solution {
    public int solution(int[][] lines) {
        int answer = 0;
        int[] coverage = new int[200];
        
        for (int i = 0; i < lines.length; i++) {
            int start = lines[i][0] + 100;
            int end = lines[i][1] + 100;
            for (int j = start; j < end; j++) {
                coverage[j]++;
            }
        }

        for (int i = 0; i < coverage.length; i++) {
            if (coverage[i] > 1) {
                answer++;
            }
        }

        return answer;
    }
}

File: 프로그래머스/0/120876. 겹치는 선분의 길이/겹치는 선분의 길이.py
def solution(lines):
    answer = 0
    coverage = [0] * 200

    for line in lines:
        start = line[0] + 100
        end = line[1] + 100
        for i in range(start, end):
            coverage[i] += 1

    for c in coverage:
        if c > 1:
            answer += 1

    return answer